### DISCUSSION TITLE
Problem with make_future_dataframe()

### DISCUSSION THREAD

**Discussion by Mo-Abdelhameed** [ourownstory/neural_prophet#1504] [Q&A - get help using NeuralProphet] [answered]
I am running the following code:

import pandas as pd
df = pd.read_csv("https://github.com/ourownstory/neuralprophet-data/raw/main/kaggle-energy/datasets/tutorial04.csv")
df.head()

from neuralprophet import NeuralProphet, set_log_level
set_log_level("ERROR")
m = NeuralProphet(
    # Disable trend changepoints
    n_changepoints=10,
    # Disable seasonality components
    yearly_seasonality=True,
    weekly_seasonality=True,
    daily_seasonality=True,
    # Add the autogression
    n_lags=10,
    epochs=1,
    n_forecasts=5
)
m.add_lagged_regressor("temperature")
metrics = m.fit(df)

df_future = m.make_future_dataframe(df=df, periods=10, n_historic_predictions=2)

The future dataframe looks like  this
<img width="536" alt="Screenshot 2024-01-09 at 2 37 49 PM" src="https://github.com/ourownstory/neural_prophet/assets/73613367/1097e9f4-e6da-4f41-877e-c895818133f1">

I am specifying periods = 10 but the future data points created is 5 which is equal to n_forecasts parameter. What is the use of periods then? Also the n_historic_predictions does not seem to be working?

When i run this line afterwards 

forecast = m.predict(df_future)

this is what i get:
<img width="1361" alt="Screenshot 2024-01-09 at 2 40 20 PM" src="https://github.com/ourownstory/neural_prophet/assets/73613367/d0583e0a-3f7e-4143-8b6f-e2921ec6b3ec">

I am having a problem understanding what the output means and why does it have NaN values. Can someone help me please?

What I want to do on general is to forecast a series and add lagged and future regressors. I have a problem with lagged regressors because i get results like the one i am showing and i do not understand what it means?

Thanks!

**Comment by rafiuddinkhan** [comment]
Yes it seems to be confusing in latest version but you can try these:

You get the latest forecast from the predicted values.

`df_forecast = m.get_latest_forecast(forecast)`

or you can try:

```
def exctract_yhat(forecasts, size):
  columns = forecasts.columns[3:]
  newframe = forecasts[['ds', 'yhat1']].iloc[-size:].copy()
  for col in columns:
      if 'yhat' in col:
         newframe['yhat1'] = newframe['yhat1'].fillna(forecasts[col])
  return newframe
```

p=exctract_yhat(forecast,5)

For details look:
https://github.com/ourownstory/neural_prophet/discussions/1541

**Comment by ourownstory** [comment, marked as answer]
Maybe this tutorial may be of help: https://neuralprophet.com/how-to-guides/feature-guides/collect_predictions.html
have a closer look at the 'raw' format.

**Comment by suntochukwu** [comment]
@Mo-Abdelhameed  could you explain or share the code on how you solved this? I'm having the same isssue.
